Makefile

```diff
@@ -122,11 +128,20 @@ endif
 gopkg:
 	@echo $(PROJECT)
 
+# Build a release tarball
+charmstore-$(VERSION).tar.xz: $(VERSIONDEPS)
+	mkdir -p charmstore-$(VERSION)/bin
+	GOBIN=$(CURDIR)/charmstore-$(VERSION)/bin go install $(INSTALL_FLAGS) -v $(PROJECT)/...
+	mv charmstore-$(VERSION)/bin/charmd charmstore-$(VERSION)/bin/charmstore
+	tar cfJv $@ charmstore-$(VERSION)
```

Comment:
I had to look up the J flag (it's not even really documented, though I can guess). I wonder about using more usual primitives:

   tar c charmstore-$(VERSION) | xz > $@

?